src/main/ApplicationUtils.cpp

```diff
@@ -25,6 +26,53 @@
 namespace stellar
 {
 
+int
+runAsRelayWithConfig(Config cfg)
+{
+    LOG(INFO) << "Starting relay-mode stellar-core " << STELLAR_CORE_VERSION;
+    VirtualClock clock(VirtualClock::REAL_TIME);
+    Application::pointer app;
+    try
+    {
+        app = Application::create<RelayApplication>(
+            clock, cfg, Application::InitialDBMode::APP_DB_DONT_OPEN);
+
+        // Relays are never validating.
+        cfg.NODE_IS_VALIDATOR = false;
```

Comment:
I think you may want to set that before creating application

Reply:
Nice catch! Thanks, fixed (that was the biggest remaining bug!)